### PR TITLE
Remove action allowlist on the server side

### PR DESCRIPTION
**Pull request by ywangd**
With the new cross-cluster API keys (#95714), RCS 2.0 is more tightly controlled. This removes the need for the action allowlist on the server side because cross-cluster API keys (1) require manage_security to create and (2) are specifized to give out only cross-cluster operation related privileges. Hence this PR removes the allowlist.

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java [hunk 1]
```diff
@@ -281,7 +246,7 @@ public void testCrossClusterAccessInboundAuthenticationException() {
         Exception authE = authenticationError("authc failed");
         // Only pick allowlisted action -- it does not make sense to pick one that isn't because we will never get to authenticate in that
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java:247]:
```suggestion        
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java [hunk 2]
```diff
@@ -281,7 +246,7 @@ public void testCrossClusterAccessInboundAuthenticationException() {
...
         // case
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java:248]:
```suggestion
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java [hunk 3]
```diff
@@ -48,7 +48,6 @@
 import static org.elasticsearch.xpack.core.security.support.Exceptions.authenticationError;
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java:48]: We gotta fix up `testInboundDestructiveOperations` as well.

**Review by n1v0lg — APPROVED** [review]
LGTM. I'm on board with this as it makes the code easier to maintain. I agree with your analysis in the Jira of the security impact (i.e., lack therefore) of this change.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java (hunk 3 above)
Comment (ywangd) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java:48]: Good catch  👍  I totally missed it

**Pull request merged**